tools/ejections/cmd/main.go

```diff
@@ -229,3 +263,308 @@ func RunScan(ctx *cli.Context) error {
+			metric.QuorumId,
+			metric.TotalUnsignedBatches,
+			metric.Percentage,
+			metric.StakePercentage,
+			metric.PerfScore,
+			metric.IsViolatingSLA,
+			metric.IsViolatingThreshold,
+			metric.NeedsEjection,
+		})
+	}
+
+	// Print the non-signing metrics table
+	fmt.Println(nonSigningTableV1.Render())
+
+	// V2 Only signing metrics
+	nonSigningMetricsV2Only := make([]*ejector.NonSignerMetric, 0)
+	if config.EvalV2 {
+		nonsigningRateV2, err := dataapiClient.GetNonSigningRateV2(time.Now(), config.EvalInterval)
+		if err != nil {
+			return fmt.Errorf("failed to get v2 nonsigning rate: %w", err)
+		}
+		for _, metric := range nonsigningRateV2.OperatorSigningInfo {
+			if _, ok := quorumIDs[core.QuorumID(metric.QuorumId)]; ok {
+				nonSigningMetricsV2Only = append(nonSigningMetricsV2Only, &ejector.NonSignerMetric{
+					OperatorId:           metric.OperatorId,
+					OperatorAddress:      metric.OperatorAddress,
+					QuorumId:             metric.QuorumId,
+					TotalUnsignedBatches: metric.TotalUnsignedBatches,
+					Percentage:           100 - metric.SigningPercentage,
+					StakePercentage:      metric.StakePercentage,
+				})
+			}
+		}
+		err = e.EvaluateOperatorsForEjection(nonSigningMetricsV2Only)
+		if err != nil {
+			return fmt.Errorf("failed to evaluate operators for v2 ejection: %w", err)
+		}
+		sort.Slice(nonSigningMetricsV2Only, func(i, j int) bool {
+			return nonSigningMetricsV2Only[i].OperatorAddress < nonSigningMetricsV2Only[j].OperatorAddress
+		})
+
+		// Create a new table writer for non-signing metrics
+		nonSigningTableV2 := table.NewWriter()
+		nonSigningTableV2.AppendHeader(table.Row{"Operator Address", "Quorum ID", "Unsigned Batches", "V2 Non Signing %", "Stake %", "Perf Score", "Violating SLA", "Violating Threshold", "Needs Ejection"}, table.RowConfig{AutoMerge: true})
+		// Set the column configuration to merge the Operator Address column
+		nonSigningTableV2.SetColumnConfigs([]table.ColumnConfig{
+			{Number: 1, AutoMerge: true}, // Merging the Operator Address column
+			{Number: 2, AutoMerge: false},
+			{Number: 3, AutoMerge: false},
+			{Number: 4, AutoMerge: false},
+			{Number: 5, AutoMerge: false},
+			{Number: 6, AutoMerge: false},
+			{Number: 7, AutoMerge: false},
+			{Number: 8, AutoMerge: false},
+			{Number: 9, AutoMerge: false},
+		})
+		nonSigningTableV2.SetStyle(table.StyleLight)
+		nonSigningTableV2.Style().Options.SeparateRows = true
+		for _, metric := range nonSigningMetricsV2Only {
+			nonSigningTableV2.AppendRow(table.Row{
+				metric.OperatorAddress,
+				metric.QuorumId,
+				metric.TotalUnsignedBatches,
+				metric.Percentage,
+				metric.StakePercentage,
+				metric.PerfScore,
+				metric.IsViolatingSLA,
+				metric.IsViolatingThreshold,
+				metric.NeedsEjection,
+			})
+		}
+		fmt.Println(nonSigningTableV2.Render())
+
+		// Merge non-signing metrics from V1 and V2
+		nonSigningMetricsMerged := e.MergeNonSigningMetrics(nonsigningRateV1, nonsigningRateV2, quorumIDs, logger)
```

Comment:
Do we need a check to call `MergeNonSigningMetrics` only in hybrid mode?
Because as it is returning metrics with worse signing rate in `v2-only` mode v1 signing rate will always be 0.

Reply:
There is no `v2-only` ejection mode. V1 signing rate is always checked. If v2-ejections are enabled, they will be included and `MergeNonSigningMetrics` will pick the worst signing rate.